### PR TITLE
[ISSUE #4261] Some code optimizations to the EventMeshCloudEventBuilder .

### LINKED ISSUE #4261: [Enhancement] Some code optimizations to the `EventMeshCloudEventBuilder` .
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
image
image
image
image
image
located at:
eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java
analysis and explanation:
1. Local variable 'eventMeshMessage' and 'cloudEvent' is redundant.[145,194]
2. Remove the parentheses around the "key" parameter[151]
3. Can be replaced with 'Map.forEach()'.[164]
4. Repeated code operations can be accomplished using a single method.[195-242]
5. Unchecked cast: 'java.util.Set<java.util.HashMap<java.lang.String,java.lang.String>>' to 'T'

### Describe the solution you'd like

1. Inline variable
2. Remove the parentheses around the "key" parameter
3. Replaced with 'Map.forEach()'.
image
4. Extract as a method.
Constants.EVENTMESH_MESSAGE_CONST_TTL is the same as ProtocolKey.TTL.
image
image
5. Add `@SuppressWarnings("unchecked")` to this method.
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by silentchildh**
Fixes #4261

### Motivation
As the title says, I'm working on some code optimizations to the `EventMeshCloudEventBuilder`.

See more details here: #4261 

### Modifications

Fixed the problem that the problem posed, and additionally fixed to get the correct IDC.

before:
image

after:
image

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by Alonexc — APPROVED** [review]
LGTM

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java [hunk 1]
```diff
@@ -190,56 +187,25 @@ private static CloudEvent switchEventMeshMessage2EventMeshCloudEvent(EventMeshMe
...
-        }
 
-        if (null == cloudEvent.getExtension(ProtocolKey.LANGUAGE)) {
-            cloudEventBuilder.withExtension(ProtocolKey.LANGUAGE, Constants.LANGUAGE_JAVA);
-        }
+        CloudEventBuilder cloudEventBuilder = CloudEventBuilder.from(message);
+
+        builderCloudEvent(message, cloudEventBuilder, ProtocolKey.ENV, clientConfig.getEnv());
+        builderCloudEvent(message, cloudEventBuilder, ProtocolKey.IDC, clientConfig.getIdc());
+        builderCloudEvent(message, cloudEventBuilder, ProtocolKey.IP, Objects.requireNonNull(IPUtils.getLocalAddress()));
```
Comment (Pil0tXia) [inline eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java:195]: How about changing the method name to `buildCloudEvent`?

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java (hunk 1 above)
Comment (silentchildh) [reply eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java]: > How about changing the method name to `buildCloudEvent`?

Yes, it does. `buildCloudEvent` seems to fit this behavior better.

**Review by Pil0tXia — APPROVED** [review]
LGTM

**Review by mxsm — DISMISSED** [review]
LGTM

File: eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java [hunk 2]
```diff
@@ -248,6 +214,12 @@ private static CloudEvent switchCloudEvent2EventMeshCloudEvent(io.cloudevents.Cl
         return null;
     }
 
+    private static void buildCloudEvent(io.cloudevents.CloudEvent message, CloudEventBuilder cloudEventBuilder, String protocolKey, String key) {
```
[the remark this answers is not in the document]
Comment (Pil0tXia) [reply eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java:217]: Reasonable. 
(Though for the moment most similar treatments are not called this name)😂.

[the remark this answers is not in the document]
Comment (silentchildh) [reply eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/grpc/util/EventMeshCloudEventBuilder.java:217]: 1. Yes, it's easier to read with the suffix `IfAbsent` .
2. But I think `build` might be more of a prefix fix than `compute`.After all, we use `CloudEventbuilder` to `Build` it.
3. Maybe we could change `String protocolKey` to `String extension` and `String key` to `String value.`.

Finally, if you have a better suggestion, I will be more than happy to see it.

**Review by mxsm — COMMENTED** [review]
@silentchildh  Please check code style

**Review by mxsm — APPROVED** [review]

**Pull request merged**